src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1857,26 +1857,16 @@ public void writeStyle( StyleInfo style, InputStream in ) throws IOException {
 	 *            file to update
 	 * @throws IOException
 	 */
-	public static void writeStyle(final InputStream in, final File styleFile)
+	public static void writeStyle(final InputStream in, final Resource styleFile)
 			throws IOException {
-		final File temporaryFile = File.createTempFile(styleFile.getName(),
-				null, styleFile.getParentFile());
 		BufferedOutputStream out = null;
 		try {
-			out = new BufferedOutputStream(new FileOutputStream(temporaryFile));
+			out = new BufferedOutputStream(styleFile.out());
 			IOUtils.copy(in, out);
 			out.flush();
 		} finally {
 			out.close();
 		}
-		// move the file
-		try {
-			org.geoserver.data.util.IOUtils.rename(temporaryFile, styleFile);
```

Comment:
@NielsCharlier I think the purpose of this renaming is to ensure that style changes are atomic filesystem operations and that other threads do not see a half-written style file during the update. Is there a way to preserve this safety within the resource API? Perhaps @aaime or @jodygarnett can advise. I am concerned that otherwise this might be a regression.